### ISSUE TITLE
When handing vlan unavailable, recalculate ready aggs

### ISSUE THREAD

**Issue by ahelsing** [GENI-NSF/geni-tools#720]
In a run Sarah did, saw this in logs:
```
There are 2 ready aggregates: [Aggregate('utah-ig')](Aggregate('ion'),)
....
10/03 16:53:32 WARNING  objects.py:2123 sliverstatus: 121251 is (still) failed at <Aggregate ion>. Delete and retry.
10/03 16:53:32 DEBUG    objects.py:2129   Status had error message: VLAN PCE(PCE_CREATE_FAILED): 'There are no VLANs available on link ion.internet2.edu:rtr.atla:ge-10/3/2:protogeni  on reservation ion.internet2.edu-121251 in VLAN PCE'
.....
10/03 16:54:23 DEBUG    objects.py:3124 deletesliver <Aggregate utah-pg> Result: Deleted sliver urn:publicid:IDN+ch.geni.net:tutorial+slice+stitch2 at utah-pg
10/03 16:54:23 INFO     launcher.py:55 Will put <Aggregate ion> back in the pool to allocate. Got: Retrying reservations at earlier AMs to avoid unavailable VLAN tag at <Aggregate ion>....
10/03 16:54:23 INFO     launcher.py:63 Pausing for 30 seconds for Aggregates to free up resources...


10/03 16:54:53 WARNING  objects.py:500 Cannot allocate at <Aggregate utah-ig>: dependencies not ready
10/03 16:54:53 DEBUG    launcher.py:48 
There are 1 ready aggregates: [Aggregate('utah-pg')]
```

Explanation:
launcher had 2 ready aggs: Utah-IG and ION.
Did Allocation at ION.
Allocation at ION had an error. This error caused us to delete the reservation at ION and at Utah-PG. 
But the code in launcher didn't re-calculate the list of 'ready' aggregates. So it tried the next ready aggregate - which was Utah-IG. So we got that warning message that Utah-IG was not ready.
The message is harmless in this case: the allocate returned, launcher fell through and re-calculated the list of ready aggregates, and then proceeded as normally. But adding a 'break' in the right spot there would avoid the ugly WARN.

 _Imported from trac ticket [#720](http://trac.gpolab.bbn.com/gcf/ticket/720),  created by ahelsing on 10-06-2014 at 10:23, last modified: 10-06-2014 at 12:43_

**Comment by ahelsing** [comment]
I have this minor fix in a branch off of release-2.7. If we end up needing RC5, I'll work it in. Otherwise I'll hold it for the next release.

_Trac comment by ahelsing on 10-06-2014 at 10:26_

**Comment by ahelsing** [comment]
There's enough else I need to do, I'm merging this in.


_Trac comment by ahelsing on 10-06-2014 at 12:43_
